lite/valor_lite/classification/manager.py

```diff
@@ -0,0 +1,854 @@
+            rocauc,
+            mean_rocauc,
+        ) = compute_metrics(
+            data=data,
+            label_metadata=label_metadata,
+            score_thresholds=np.array(score_thresholds),
+            hardmax=hardmax,
+            n_datums=n_datums,
+        )
+
+        metrics = defaultdict(list)
+
+        metrics[MetricType.ROCAUC] = [
+            ROCAUC(
+                value=rocauc[label_idx],
+                label=self.index_to_label[label_idx],
+            )
+            for label_idx in range(label_metadata.shape[0])
+            if label_metadata[label_idx, 0] > 0
+        ]
+
+        metrics[MetricType.mROCAUC] = [
+            mROCAUC(
+                value=mean_rocauc[label_key_idx],
+                label_key=self.index_to_label_key[label_key_idx],
+            )
+            for label_key_idx in range(len(self.label_key_to_index))
+        ]
+
+        for label_idx, label in self.index_to_label.items():
+
+            kwargs = {
+                "label": label,
+                "score_thresholds": score_thresholds,
+                "hardmax": hardmax,
+            }
+            row = counts[:, label_idx]
+            metrics[MetricType.Counts].append(
+                Counts(
+                    tp=row[:, 0].tolist(),
+                    fp=row[:, 1].tolist(),
+                    fn=row[:, 2].tolist(),
+                    tn=row[:, 3].tolist(),
+                    **kwargs,
+                )
+            )
+
+            # if no groundtruths exists for a label, skip it.
+            if label_metadata[label_idx, 0] == 0:
+                continue
+
+            metrics[MetricType.Precision].append(
+                Precision(
+                    value=precision[:, label_idx].tolist(),
+                    **kwargs,
+                )
+            )
+            metrics[MetricType.Recall].append(
+                Recall(
+                    value=recall[:, label_idx].tolist(),
+                    **kwargs,
+                )
+            )
+            metrics[MetricType.Accuracy].append(
+                Accuracy(
+                    value=accuracy[:, label_idx].tolist(),
+                    **kwargs,
+                )
+            )
+            metrics[MetricType.F1].append(
+                F1(
+                    value=f1_score[:, label_idx].tolist(),
+                    **kwargs,
+                )
+            )
+
+        if MetricType.ConfusionMatrix in metrics_to_return:
+            metrics[
+                MetricType.ConfusionMatrix
+            ] = self._compute_confusion_matrix(
+                data=data,
+                label_metadata=label_metadata,
+                score_thresholds=score_thresholds,
+                hardmax=hardmax,
+                number_of_examples=number_of_examples,
+            )
+
+        for metric in set(metrics.keys()):
+            if metric not in metrics_to_return:
+                del metrics[metric]
+
+        return metrics
```

Comment:
can you return an array of dictionaries for each metric key, instead of a dictionary of metric objects? that would help resolve this ticket: https://github.com/orgs/Striveworks/projects/2/views/1?pane=issue&itemId=80550739

Reply:
Im not sure what you mean in this ticket, but the output currently matches obj detection so im going to consider this suggestion out of scope for this PR.

Reply:
we'd discussed not returning metric objects to the user since they'd just have to convert them to dictionaries, anyway. not considering this OOS

Reply:
Added the `as_dict` arg to the `evaluate` call to provide the requested behavior.